### ISSUE TITLE
Property without a setter, typescript isn't catching error when settting

### ISSUE THREAD

**Issue by RoyM33** [Microsoft/TypeScript#1810]
if you create a "property" with a public getter and no setter. if you then try to set that property typescript doesn't tell you this isn't allowed. But it sure will blow up your code.

        private _date: Date;
        public get Date()
        {
            return this._date;
        }

In the above scenario if you do
      this.Date = new Date();
typescript allows this without catching anything (no red underbar or error), but the javascript doesn't allow it. Changing the code to the below code works as expected.
       this._date = new Date();

**Comment by danquirk** [comment]
See #12

**Comment by buzinas** [comment]
In fact #12 and this one is completely different.

The other is talking about readonly properties in interfaces (something that the current **language** does not support). This one is talking about a simple validation to throw a compiler error when trying to change a getter only property.

This one is very simple to implement (and, for me, it's a must), totally different from the other.

**Comment by RyanCavanaugh** [comment]
"Property without a setter" and "Property that is read-only" is a distinction without a difference.

For example, if we have some class
```ts
class Foo {
  get bar() { return 0 }
}
```
This is its corresponding .d.ts:
```ts
class Foo {
  bar: number;
}
```
Without a `readonly` modifier, there's no way to represent this *except* when we can see the implementation. Trying to implement this as via "see if there's a setter" would be a hack.
